mat64/io.go

```diff
@@ -5,14 +5,152 @@
 package mat64
 
 import (
+	"bytes"
 	"encoding/binary"
+
+	"github.com/gonum/matrix"
 )
 
 var (
-	littleEndian  = binary.LittleEndian
-	bigEndian     = binary.BigEndian
-	defaultEndian = littleEndian
-
+	binLE       = binary.LittleEndian
 	sizeInt64   = binary.Size(int64(0))
 	sizeFloat64 = binary.Size(float64(0))
 )
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Dense is little-endian encoded as follows:
+//   0 -  8  number of rows    (int64)
```

Comment:
Agree with @vladimir-ch, make these 0 - 7 etc. This is a format doc, not a mathematics statement.